src/core/environment.js

```diff
@@ -737,4 +737,80 @@ p5.prototype.getURLParams = function() {
   return v;
 };
 
+/**
+ * Reduces the log load to get the better performance and
+ * more informative logs from console, by ignoring repeated values or setting periods.
+ * Instead of using console.log(a) to display the value of variable a,
+ * now you may use lazyLog('a'). Calling the function without any parameters will
+ * clear the cache of logged variables.
+ *
+ * @method lazyLog
+ * @param {Object} variable can be any JavaScript data type
+ * @param {String} variableName a tag to help cache different variables, can just be the same as the variable name
+ * @param {Number} [period] number of frames (calls of the function) not logging
+ * @param {Boolean} [display] set to false if not displaying the name of variable in log
+ * @return {Boolean} true if the variable changes or match the logging condition, otherwise false
+ * @example
+ * <div>
+ * <code>
+ * // Log only once when the value stays unchanged
+ * let x = 50;
+ * function draw() {
+ *   ellipse(x, 50, 50);
+ *   lazyLog(x, 'x');
+ * }
+ * </code>
+ * </div>
+ *
+ * <div>
+ * <code>
+ * // Log frame rate
+ * function draw() {
+ *   lazyLog(int(getFrameRate()), 'fr');
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * Log only once when the value stays unchanged.
+ * Log frame rate.
+ */
+/**
+ * @method lazyLog
+ */
+p5.prototype.lazyLog = function(...args) {
+  let argsLength = args.length;
+  let displayName = true;
+
+  if (!argsLength) {
+    // Clear the cached variables
```

Comment:
is clearing when no args are passed a common pattern? having a separate function or explicit parameter for this feels like it would be safer.

Reply:
Hi @dhoizner, thanks for pointing it out. I opened this pull request before noticing the ongoing discussion around "adding new features only if accessibility increased," and I hope this function could help people access the library better by providing less while more useful and concentrated information in the console, with less computing resources required. With that being said, I also want the interface of it to be as simple as possible. How do you think about `lazyLog(CLEAR)` or a separate `clearLog()`/`clearLazyLog()` function?

Reply:
hi @peilingjiang i think this would make debugging a lot easier, so i'm all for bringing this in (although i'm not one of the maintainers!). i think that having a `clearLazyLog` makes the most sense to me.